kolibri/core/content/api.py

```diff
@@ -375,7 +386,31 @@ def filter_options(self, request, **kwargs):
 
 @method_decorator(remote_metadata_cache, name="dispatch")
 class ChannelMetadataViewSet(BaseChannelMetadataMixin, RemoteViewSet):
-    pass
+    # Overriding to isolate Public API from internal API
+    values = tuple(
+        field for field in BaseChannelMetadataMixin.values if field != "thumbnail"
+    )
+
```

Comment:
So you would add here:
```
field_map = {
    "thumbnail": _create_channel_thumbnail_url,
}

field_map.update(BaseChannelMetadataMixin.field_map)
```

Then in the module scope you can define this function:

```
def _create_channel_thumbnail_url(item):
    return reverse("kolibri:core:channel-thumbnail", args=[item["id"]])
```

Reply:
This is not a Django thing - so the documentation for it is in this code comment: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/api.py#L139